src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -52,40 +63,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } catch (GoogleJsonResponseException e) { // some other problem with requesting comment data
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, COMMENTS_FAILED_ERROR);
     }
-    String comments = String.join(". ", commentsList);
 
     String captions = captionService.getCaptionFromId(videoId);
 
-    if (comments.isEmpty() && captions.isEmpty()) {
-      VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-          .setScore(null)
-          .setScoreAvailable(false)
-          .build();
-      String json = new Gson().toJson(videoAnalysis);
-
+    if (commentsList.isEmpty() && captions.isEmpty()) {
+      String json = createResponseJson(null, false);
       response.setContentType("application/json;");
       response.getWriter().println(json);
       return;
     }
+    
+    // Creates 26 thread pools (25 for comments, 1 for captions)
+    ExecutorService executor = Executors.newFixedThreadPool(26);
 
-    String text = comments + ". " + captions;
+    List<Future> commentScoreFutures = new ArrayList<>();
+    for (String comment : commentsList) {
+      Future<Float> future = executor.submit(new Callable<Float>() {
+        @Override
+        public Float call() {
+          return calculateSentimentScore(comment);
+        }
+      });
+      commentScoreFutures.add(future);
+    }
```

Comment:
You can simplify this to a single line using a method reference and stream.map. Something like:

```
List<Future<Float>> commentScores = commentsList.stream().map(this::calculateSentimentScore).collect(Collectors.toList());
```

Reply:
Cool, I've never used Java Streams before! I couldn't really figure out how to use ```this::calculateSentimentScore``` because I had to use Callable. Ended up using a lambda expression ```comment -> executor.submit(callableFactory(comment))``` and made a callableFactory method that calls calculateSentimentScore from there

Reply:
Hm, I thought you should be able to pass a lambda to a method that expects a Callable. and you should be able to use a method reference anywhere you can pass a lambda. But it's been a while since I've written java so I might be misremembering